src/OpenConext/UserLifecycle/Domain/ValueObject/InactivityPeriod.php

```diff
@@ -19,28 +21,22 @@
 namespace OpenConext\UserLifecycle\Domain\ValueObject;
 
 use OpenConext\UserLifecycle\Domain\Exception\InvalidInactivityPeriodException;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Webmozart\Assert\Assert;
 
 class InactivityPeriod
 {
-    /**
-     * @var int
-     */
-    private $period;
 
-    public function __construct($inactivityPeriod)
-    {
-        if (!is_int($inactivityPeriod) || $inactivityPeriod <= 0) {
-            throw new InvalidInactivityPeriodException('The inactivity period must be an integer value');
-        }
+    public function __construct(
+        #[Autowire(param: 'inactivity_period')]
+        private readonly int $inactivityPeriodInMonths,
+    ) {
+        Assert::greaterThan($inactivityPeriodInMonths, 0, 'The inactivity period must be an positive integer value');
```

Comment:
I refrained from using a third party package in this VO as we do not want to add coupling to infra/3rd party dependencies in the domain layer.

Reply:
I will remove it then, did not realise this was a requirement.

Reply:
Thanks. We might have to re consider this. As having rigid type checks is a very valuable thing. On the other hand, denying coupling with the infra layer from the domain layer is important too.

In the past, these type checks would be written using phpdoc annotations. Now that they are an actual part of the PHP language with the introduction of `#Attributes` we might need to re-consider our standpoint.

Reply:
So what do you suggest? Using  Symfony's `#[Assert\Positive]` here?

Reply:
The checking is as rigid as with the Assert function of webmozart, but without the 3rd party dependency). Looks like a win/win to me?

Reply:
Yeah this is purfect :cat2: